fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1586,9 +1574,76 @@ public static IndexState writeOnlyIfTooManyRecordsForRebuild(long recordCount, b
     }
 
     @Nonnull
-    @API(API.Status.INTERNAL)
-    protected CompletableFuture<KeyValue> readStoreFirstKey() {
-        final AsyncIterator<KeyValue> iterator = context.ensureActive().getRange(getSubspace().range(), 1).iterator();
+    private static RecordMetaDataProto.DataStoreInfo parseDataStoreInfo(@Nullable KeyValue keyValue, @Nonnull FDBRecordContext context,
```

Comment:
Maybe the method or the first argument could use a slightly more descriptive name to make it clear that this checks the given key value and parses (and returns) it if it's the store info and does other interesting things if it isn't.